sql/set.go

```diff
@@ -39,7 +39,12 @@ func (p *planner) Set(n *parser.Set) (planNode, error) {
 		if err != nil {
 			return nil, err
 		}
-		p.session.Database = val.String()
+		if s, ok := val.(parser.DString); ok {
+			p.session.Database = string(s)
```

Comment:
`s` has the type `parser.DString` which is `type DString string`. So this is a copy of the `string` struct, but not the underlying pointer data.